# Data Analysis of AppStore and PlayStore apps

This project is a practise project for data analysis.
We are going to sort through a data set containing lists of apps and corresponding information and apply Pythin techniques to understand what kind of apps are likely to attract more users on the PlayStore. <br>
We are making use of two datasets, one for the Apple Playstore and one for the Google Playstore. <br>
The Apple AppStore dataset can be found [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps). <br>
The Google PlayStore dataset can be found [here](https://www.kaggle.com/lava18/google-play-store-apps).
<br><br>


In [1]:
from csv import reader
playstore_dataset=open('googleplaystore.csv',encoding='utf8')
read_file=reader(playstore_dataset)
playstore=list(read_file)
playstore_header=playstore[0]
playstore_apps=playstore[1:]

applestore_dataset=open('AppleStore.csv', encoding='utf8')
read_file=reader(applestore_dataset)
applestore=list(read_file)
applestore_header=applestore[0]
applestore_apps=applestore[1:]

First we create a function that allows us to explore the datasets. This can help identify the rows and columns of the dataset and help us decide which parts of the data we can use for further analysis.

In [2]:
def explore_data(dataset,start,end,length=False):
    dataset_new=dataset[start:end]
    for row in dataset_new:
        print(row)
        print('\n')
    if length==True:
        print('The number of rows is: ',len(dataset))
        print('The number of columns is ',len(dataset[0]))

In [3]:
print('This is a glimpse of the Apple AppStore dataset: \n')
explore_data(applestore, 0,6)
print('\n\n\n')
print('This is a glimpse of the Google PlayStore dataset: \n')
explore_data(playstore,0,6)

This is a glimpse of the Apple AppStore dataset: 

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']






Thi

There seems to be an error in one of the entries in the PlayStore dataset as shown in the discussion page [here](https://www.kaggle.com/lava18/google-play-store-apps/discussion/81460). <br>
One possible solution is correct the data. But what I have done is simply deleted that entry.<br>
Make a note that the below  cell should not be run more than once as it can lead to deletetion of multiple rows, which is undesirable.

In [4]:
print(playstore_header)
print('\n')
print(playstore_apps[10472])
print('\n Here we see that the app has a rating of 19, which is not possible.')
print('There is an error in the index 10472 of the PlayStore dataset and so we delete that entry.')


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']

 Here we see that the app has a rating of 19, which is not possible.
There is an error in the index 10472 of the PlayStore dataset and so we delete that entry.


In [5]:
print("The length of the dataset before deletion is : ",len(playstore_apps))
del playstore_apps[10472]
print("The length of the dataset afterr deletion is : ", len(playstore_apps), ", as expected.")
print("The new entry at index 10472 is now- \n")
print(playstore_apps[10472])

The length of the dataset before deletion is :  10841
The length of the dataset afterr deletion is :  10840 , as expected.
The new entry at index 10472 is now- 

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


# Finding and Deleting duplicate entries

On inspection of the dataset, it can be seen that there are a lot of duplicate entries. <br>
For e.g, Instagram has 4 entries as shown below - 

In [6]:
count=0
for app in playstore_apps:
    if app[0]=="Instagram":
        print(app)
        count+=1
print("\n Here we see that Instagram has", count,"entries in the dataset")

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']

 Here we see that Instagram has 4 entries in the dataset


Similarly, we can find many other instances of duplicate entries.

In [7]:
unique_apps=[]
duplicate_apps=[]
for app in playstore_apps:
    if app[0] in unique_apps:         #the index 0 corresponds to the name of the app
        duplicate_apps.append(app[0])
    else:
        unique_apps.append(app[0])
print("\n There are", len(duplicate_apps), "duplicate entries in the dataset.")
print("Hence the actual length of the PlayStore dataset is",len(playstore_apps)-len(duplicate_apps))


 There are 1181 duplicate entries in the dataset.
Hence the actual length of the PlayStore dataset is 9659


## Removing duplicates
We will not randomly be removing all duplicates. <br>
As the above cell shows, all the 4 "Instagram" instances have different number of reviews (index number 3). The one with the most number of reviews can be safely assumed to be the latest entry to the dataset and hence we will keep  that one.
So, we need to separate all sets  of duplicates and keep only those which have the most number of reviews. <br> <br>
The approach we have chosen is - <br>
1. An empty dictionary `reviews_max` is created, which is used to store the max value of number of reviews corresponding to each app. <br>
2. Iterating through every app in the PlayStore app list, assign `name` and `n_reviews` and check if `name` is already present in `reviews_max`, and if `n_reviews` is greater than the number of reviews in the dictionary entry. If these two conditions are satisfied then replace the value of `name` in the dictionary with `n_reviews`. <br>
3. If `name` is not in `reviews_max`, then create a dictionay entry with key = `name` and value = `n_reviews`. <br>

Once this is done, we can see that the number of elements in the dictionary is 9659, as expected. <br>
Now we can move on to creating a list of apps with the duplicates removed.
<br>
1. Create two empty lists, one to store the app details and one to store only the names. <br>
2. Iterating through the PlayStore app list, check if each app is not present in the names-only list and if that entry contains the maximum number of reviews for that app by using the dictionary entry. If these are satisfied then append the whole row into the other list. Also append the name into the names list, as it helps keep track of the app list.

In [8]:
reviews_max={}
for app in playstore_apps:
    name=app[0]
    n_reviews=float(app[3])
    if name in reviews_max and n_reviews>reviews_max[name]:
        reviews_max[name]=n_reviews
    if name not in reviews_max:
        reviews_max[name]=n_reviews

android_clean=[]
already_added=[]           #names only
for app in playstore_apps:
    name=app[0]
    n_reviews=float(app[3])
    if n_reviews== reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(name)

print("The length of the new andoird_clean list is",len(android_clean),", as expected.")

        

The length of the new andoird_clean list is 9659 , as expected.


Checking the AppleStrore dataset reveals that no duplicates are present.

In [9]:
unique_apps=[]
duplicate_apps=[]
for app in applestore_apps:
    if app[0] in unique_apps:         #the index 0 corresponds to the name of the app
        duplicate_apps.append(app[0])
    else:
        unique_apps.append(app[0])
print("\n There are", len(duplicate_apps), "duplicate entries in the dataset.")


 There are 0 duplicate entries in the dataset.


Hence we can assign appropriate names to the two proper lists.

In [10]:
android=android_clean
ios=applestore_apps

Henceforth, we will use the above lists for accessing app data.

## Checking the names of apps
This project is targeted towards an English audience and so we would like the dataset to have apps with English names. But going through the datasets, we can see that there are several apps with names conatining non-English charecters.

In [11]:
print(applestore_apps[813][1])
print(applestore_apps[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


We do not need these apps in our dataset as they will yeild unwanted results. <br>
To remove these from the dataset, we first need to find ways to detect a non-English name. For this we use the in-built `ord()` function which returns the ASCII number for the corresponding character. All of our English alphabet and commonly used symbols and numbers all have an ASCII number less than 127. So, using this criteria we can find out if the name of each app has a non-English criteria or not. <br>
We write a function which returns `False` if the string provided as argument has non-English characters, and returns `True` otherwise. Using this function we can separate the non-English oriented apps from the English ones.

In [12]:
def is_English(string):
    for char in string:
        if ord(char)>127:
            return False
    return True

print(is_English("Facebook"))
print(is_English("爱奇艺PPS"))
print(is_English('Docs To Go™ Free Office Suite'))
print(is_English('Instachat 😜'))

True
False
False
False


We see that our functions works as expected for `"Facebook"` and `"爱奇艺PPS"`, but it returns `False` for `'Docs To Go™ Free Office Suite'` and `"Instachat 😜` eventhough they contain English characters. This is so because these names contain some special characters like ™ and 😜 which have ASCII numbers more than 127. <br>
So we need to be careful while filtering through the app list using this function as we might end up loosig a lot of data because of these special characters. To avoid this we set a limit of 3 non-English characters to be present in an apps name for it to be classified as an English oriented app. So we have to modify the `is_English` function.

In [13]:
def is_English(string):
    count=0
    for char in string:
        if ord(char)>127:
            count+=1
        if count>3:
            return False
    return True
print(is_English('Instachat 😜'))
print(is_English('Docs To Go™ Free Office Suite'))

True
True


Now we loop through each dataset and add the English apps to another separate list.

In [14]:
android_english=[]
apple_english=[]

for app in android:                             #name is index 0 here
    if is_English(app[0])==True:
        android_english.append(app)

for app in ios:                                 #name is index 1 here
    if is_English(app[1])==True:
        apple_english.append(app)

explore_data(android_english,1,4,True)
print('\n')
explore_data(apple_english,1,4,True)

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


The number of rows is:  9614
The number of columns is  13


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9

## Isolating free apps
The last stage in preparing our dataset is to isolate the free apps into a separate list. <br>
We do this because it is the free apps that we wish to target, which has ad-based revenue.


In [15]:
android_final=[]
apple_final=[]

for app in android_english:
    if app[6]=='Free':
        android_final.append(app)

for app in apple_english:
    if float(app[4])==0.0:
        apple_final.append(app)

explore_data(android_final,0,4,True)
print('\n')
explore_data(apple_final,0,4,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


The number of rows is:  8863
The number of columns is  13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0

# Data Analysis
So far we have cleanded by the data in the dataset by-
- Removing duplicate entries
- Removing Non-English apps
- Isolating the free apps

Now we can use this dataset to perform our required analysis. <br>
The main aim here is to determine the kind of apps that are likely to attract more customers because ad-based revenue depends largely on the number of people using the app. <br>
A well accepted strategy to develop an app idea is-
1. Develop a minimal version of the app and put it on the PlayStore.
2. If the minimal version shows good response, develop  it further.
3. If the app shows profitability after around 6 months, make an iOS version and put it up on the Apple AppStore.
<br>
Since our end goal is to put up the app in both Google and Apple stores, we need to find app profiles that can be successfull in both markets. <br><br>
To begin our analysis, we need to build frequency tables for various columns, and do this we require the use of a function.

In [16]:
def freq_table(dataset,index):
    dictionary={}
    sum=0
    for app in dataset:
        if app[index] in dictionary:
            dictionary[app[index]]+=1
            sum+=1
        else:
            dictionary[app[index]]=1
            sum+=1
    for app in dictionary:
        dictionary[app]=(dictionary[app]/sum)*100
    return dictionary        

Now we define a function to display the values of a frequecny table in descending order. The function takes two parameters, `dataset` and `index` and displays the corresponding frequency table.

In [17]:
def display_table(dataset,index):
    table=freq_table(dataset,index)
    table_display=[]
    for key in table:
        tuple_form=(table[key],key)
        table_display.append(tuple_form)
    sorted_table=sorted(table_display,reverse=True)
    for value in sorted_table:
        print(value[1],":",value[0])

Now we use this function to analyse the two datasets. <br>
We begin by looking into the Apple app list. The header row reveals that the index number 11 has information about each apps genre, so we use the `display_table()` function and look at how varoius apps are distributed across different genres.

In [18]:
display_table(apple_final,11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In the AppStore, 58% of the apps fall under `Games`. This is followed by an 8% in the `Entertainment` section. Following this is the category `Photo & Video` with a 5% share. <br>
The genreal observation is that most apps in the Apple AppStore are those used fro fun (Entertainment, Games, Social Media, etc). Apps for education, shopping, business and other such applications are rare.
<br>
<br>
We continue by looking into the Genres and Category columns in the PlayStore dataset.

In [19]:
display_table(android_final,1)

FAMILY : 18.898792733837304
GAME : 9.725826469592688
TOOLS : 8.462146000225657
BUSINESS : 4.592124562789123
LIFESTYLE : 3.9038700214374367
PRODUCTIVITY : 3.8925871601038025
FINANCE : 3.7007785174320205
MEDICAL : 3.5315355974275078
SPORTS : 3.396141261423897
PERSONALIZATION : 3.317161232088458
COMMUNICATION : 3.2381812027530184
HEALTH_AND_FITNESS : 3.0802211440821394
PHOTOGRAPHY : 2.944826808078529
NEWS_AND_MAGAZINES : 2.798149610741284
SOCIAL : 2.6627552747376737
TRAVEL_AND_LOCAL : 2.335552296062281
SHOPPING : 2.245289405393208
BOOKS_AND_REFERENCE : 2.1437436533904997
DATING : 1.8616721200496444
VIDEO_PLAYERS : 1.7939749520478394
MAPS_AND_NAVIGATION : 1.399074805370642
FOOD_AND_DRINK : 1.241114746699763
EDUCATION : 1.1621347173643235
ENTERTAINMENT : 0.9590432133589079
LIBRARIES_AND_DEMO : 0.9364774906916393
AUTO_AND_VEHICLES : 0.9251946293580051
HOUSE_AND_HOME : 0.8236488773552973
WEATHER : 0.8010831546880289
EVENTS : 0.7108202640189552
PARENTING : 0.6544059573507841
ART_AND_DESIGN : 0

The observation here is that a majority of the apps are Family oriented, followed by Games in second place. But looking into the Family section on the PlayStore we see that the  family categpry is just filled with games for children. <br>
Most of the apps in the PlayStore are non-game, i.e., they are all practical use apps (30-70 share for game-nongame)

In [20]:
display_table(android_final,9)

Tools : 8.450863138892023
Entertainment : 6.070179397495204
Education : 5.348076272142616
Business : 4.592124562789123
Productivity : 3.8925871601038025
Lifestyle : 3.8925871601038025
Finance : 3.7007785174320205
Medical : 3.5315355974275078
Sports : 3.463838429425702
Personalization : 3.317161232088458
Communication : 3.2381812027530184
Action : 3.102786866749408
Health & Fitness : 3.0802211440821394
Photography : 2.944826808078529
News & Magazines : 2.798149610741284
Social : 2.6627552747376737
Travel & Local : 2.324269434728647
Shopping : 2.245289405393208
Books & Reference : 2.1437436533904997
Simulation : 2.042197901387792
Dating : 1.8616721200496444
Arcade : 1.8503892587160102
Video Players & Editors : 1.771409229380571
Casual : 1.7601263680469368
Maps & Navigation : 1.399074805370642
Food & Drink : 1.241114746699763
Puzzle : 1.128286133363421
Racing : 0.9928917973598104
Role Playing : 0.9364774906916393
Libraries & Demo : 0.9364774906916393
Auto & Vehicles : 0.9251946293580051
S

The Category column is more detailed compared to the Genre column. There is more diversification among apps belonging to the same Genre as well. With a good 8.5%, Tools takes the top place for the most popular Category of apps. Entertainment apps come next with a 6%, followed by Educational apps at a 5.3%.
<br><br>
After looking at both these datasets we can say that the AppStore is dominated by apps made for fun while the PlayStore is more evenly distributed among fun and practical apps.
<br><br>
## Average installs in the AppStore
To get a better idea of which genre is the most popular, we need to calculate the average number of installs for each of the dataset. That can be done by using the `Installs` column in the adnroid dataset.<br>
The Apple dataset doesn't have a direct column for number of installs but we can use the number of reviews as a good alternative, as it is a similar measure.\

In [21]:
prime_genre_table=freq_table(apple_final,-5)
for genre in prime_genre_table:
    total=0
    len_genre=0
    for app in apple_final:
        app_genre=app[11]
        if genre==app_genre:
            n_ratings=float(app[5])
            total+=n_ratings
            len_genre+=1
    average_rating=total/len_genre
    print(genre,":",average_rating)


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


We observe that Social Networking apps have the highest average rating. This makes sense because apps like Facebook, Instagram and Snapchat have taken over the social media spcace. Almost every person with a smartphone has one or more of these apps on it. Let us look at the number of downloads for these apps individually to see why Social Media is the top category.

In [22]:
social_media_apps=[]
utilities_apps=[]

print("For the Social Networking category: \n")
for app in apple_final:
    genre=app[11]
    if genre=="Social Networking":
        app_tuple=(app[5],app[1])
        social_media_apps.append(app_tuple)
social_media_apps_sorted=sorted(social_media_apps, reverse=True)
print("(number of ratings, App) \n")
explore_data(social_media_apps,0,3,True)

print("\n For the Utilities category: \n")
for app in apple_final:
    genre=app[11]
    if genre=="Utilities":
        app_tuple=(app[5],app[1])
        utilities_apps.append(app_tuple)
utilities_apps_sorted=sorted(utilities_apps, reverse=True)
print("(number of ratings, App) \n")
explore_data(utilities_apps,0,3,True)


For the Social Networking category: 

(number of ratings, App) 

('2974676', 'Facebook')


('1061624', 'Pinterest')


('373519', 'Skype for iPhone')


The number of rows is:  106
The number of columns is  2

 For the Utilities category: 

(number of ratings, App) 

('479440', 'Google – Search made just for mobile')


('130450', 'Flashlight Ⓞ')


('126948', 'My Verizon')


The number of rows is:  81
The number of columns is  2


Facebook and Pinterest top the Social Media list with 1000000+ ratings for each. There is the possibility that these could have skewed the results in their favour, but there is no denying that Social Media is a very believable genre to be at the top in the Apple AppStore.
But since we are looking to make a new app, going this way may not be helpful as people will not be willing to shift from one social media app to another that easily. So we need to look for a category which is dynamic, i.e. nobody should have to leave another app in this category to move into this one. The category that fits this profile can be `Games`, `Referece`, `Utilities` and other similar practically oriented categories. <br>
Creating a new game is a nice way to go, but doesn't guarentee success. There are millions of games already on the AppStore and to make a new one stand out is not very easy. Creating an app for the `Reference` category, like a dictionary app or a Bible app can be done, but even here success is not a 100% sure. Getting people to switch from one dictionary to the other, eventhoiugh both contain the same words and meanings, is a Herculian task. The same goes for apps containing religious texts.
The `Utilities` category, however, shows a lot of promise. A lot of experimenting can be done in this category, with respect ro code. The top Utilities on the AppStore are Google Search and A flashlight app. A compass, ruler, calculator, convertor, etc all fall in this category, the list is endless. Our best bet would be to create an app in this Category.

## Moving on to the Google PlayStore
We saw that Social Media dominates the AppStore. Now we analyse the Google PlayStore data.This dataset has a column called `Installs` which gives the number of installs. But this data is not precise enough as it has values of the form 10000+, 1000+, etc. So we do not accurately know the number of installs for each app.

In [23]:
n_install_android=[]
for app in android_final:
    installs=app[5]
    name=app[0]
    app_tuple=(installs,name)
    n_install_android.append(app_tuple)
n_install_android=sorted(n_install_android, reverse=True)
for app_tuple in n_install_android[0:4]:
    print(app_tuple[1],":",app_tuple[0])

imo free video calls and chat : 500,000,000+
Viber Messenger : 500,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Twitter : 500,000,000+


Since we have no other choice, we use the data that we have. We will be treating "100,000+" as 100,000, "50,000+" as 50,000 and so on. <br>
To do this we need to modify the string "100,000+" into "100,00" by removing the extra symbols. The `.replace(old,new)` method can help us do this.

In [24]:
android_category=freq_table(android_final,9)

for category in android_category:
    total=0
    len_category=0
    for app in android_final:
        app_category=app[9]
        if category==app_category:
            n_install=app[5]
            n_install=int((n_install.replace('+','').replace(',','')))
            total+=n_install
            len_category+=1
    average_installs=total/len_category
    print(category,":",average_installs)

Art & Design : 2122850.9433962265
Art & Design;Creativity : 285000.0
Auto & Vehicles : 647317.8170731707
Beauty : 513151.88679245283
Books & Reference : 8767811.894736841
Business : 1712290.1474201474
Comics : 831873.1481481482
Comics;Creativity : 50000.0
Communication : 38456119.167247385
Dating : 854028.8303030303
Education : 550185.4430379746
Education;Creativity : 2875000.0
Education;Education : 4759517.0
Education;Pretend Play : 1800000.0
Education;Brain Games : 5333333.333333333
Entertainment : 5602792.775092937
Entertainment;Brain Games : 3314285.714285714
Entertainment;Creativity : 4000000.0
Entertainment;Music & Video : 6413333.333333333
Events : 253542.22222222222
Finance : 1387692.475609756
Food & Drink : 1924897.7363636363
Health & Fitness : 4188821.9853479853
House & Home : 1331540.5616438356
Libraries & Demo : 638503.734939759
Lifestyle : 1412998.3449275363
Lifestyle;Pretend Play : 10000000.0
Card : 3815462.5
Arcade : 22888365.48780488
Puzzle : 8302861.91
Racing : 1591064

The Art & Design category is at the top with 2122850 as the average number of installs. The second place category is also Art & Design but with the added Creativity constraint.

In [25]:
art=[]

print("For the Art & Design category: \n")
for app in android_final:
    category=app[9]
    if category=="Art & Design":
        art.append((app[5],app[0]))
art_sorted=sorted(art,reverse=True)
explore_data(art_sorted,0,3)

print('\n')
print("For the Art & Design; Creativity category: \n")
for app in android_final:
    category=app[9]
    if category=="Art & Design;Creativity":
        art.append((app[5],app[0]))
art_sorted=sorted(art,reverse=True)
explore_data(art_sorted,0,3)

For the Art & Design category: 

('500,000+', 'Superheroes Wallpapers | 4K Backgrounds')


('500,000+', 'Photo Designer - Write your name with shapes')


('500,000+', 'How To Color Disney Princess - Coloring Pages')




For the Art & Design; Creativity category: 

('500,000+', 'UNICORN - Color By Number & Pixel Art Coloring')


('500,000+', 'Superheroes Wallpapers | 4K Backgrounds')


('500,000+', 'Photo Designer - Write your name with shapes')




Both these categories contain similar kind of apps. Not only these, exploring every other category will reveal the same thing; most apps here are targeted for children. This makes sense because for most children their first devices are usually an Android one and come with a Google PlayStore. Taregtting this audience is not very easy as children are found to jump from one app to the other fairly quickly.

In [26]:
business=[]
for app in android_final:
    category=app[9]
    if category=="Business":
        business.append((app[5],app[0]))
business_sorted=sorted(business,reverse=True)
explore_data(business_sorted,0,3)

('500,000+', 'TANGEDCO Mobile App (Official)')


('500,000+', 'Resume PDF Maker / CV Builder')


('500,000+', 'MailChimp - Email, Marketing Automation')




Looking into the Business category, the most popular apps are a resume builder and an Email app. Making a newer app in this category can seem like a good idea but it will lead to problems later on. Let me explain how. A business app is usually targeted towards corporations and other such eshtablishments, who are always looking for cheap options. Let's suppose we created an email app, for free initially, and put it up on the PlayStore. Many corportations will incorporate this app into their eshtabloishments as it is free. But after a while, you will have to charge a fee to keep the app sustaininable. All the corporations will then start looking for alternate ways, ways in which they can void paying you this fee. This will lead them to newer free apps on the PlayStore, as there is no shortage in that regard.
<br>Making an app a sustainable one in this category can only be acheived if corporations cannot survive without the app. This can be apps like Cloud Storage, in which switching from one to another is not an easy task. Most busnisses are always looking to expand their data storage capabilities, with digital information becoming more and more popular. Creating a new Cloud Storage app may be expensive, but the payoff will be bigger once we secure loyal customers.

# Conclusions
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets. <br>
In the Apple AppStore, we saw that creating an app in the `Utilities` category is most profiatble way as there is no constraint on the specifics of what the app should do. On the Google PlayStore, we concluded that a Business app such as that of a Cloud Storage for corporations is the best way to make money.